report_thesis/src/sections/background/ensemble_learning_models/stacked_generalization.tex

```diff
@@ -1,9 +1,7 @@
-\subsection{Stacked Generalization}
-Stacked generalization, introduced by \citet{wolpertstacked_1992}, is a method designed to improve the predictive performance of machine learning models by leveraging the strengths of multiple models.
-
-In this technique, multiple base models are trained on the original dataset.
-The outputs of these base models serve as inputs to a meta-model, which is trained to make the final prediction.
-This strategy enables the meta-model to learn the optimal way to combine the outputs of the base models to minimize the generalization error.
+\subsubsection{Stacked Generalization}
+Stacked generalization, introduced by \citet{wolpertstacked_1992}, is an ensemble method that combines the predictions of multiple base models, which are trained on the original dataset, as input to a meta-model.
+This meta-model is trained to make the final prediction.
+The strategy allows the meta-model to learn the optimal way to combine the predictions of the base models to minimize the generalization error.
```

Comment:
I think some formalization would help the readability of this. "Original dataset" seems a bit unclear, and the exact variables being used for various processes isn't entirely clear.

I was thinking of some formalized version like this? The math isn't particularly advanced, so hopefully it only makes it clearer?

## Stacked Generalization

Stacked generalization, introduced by \citet{wolpertstacked_1992}, is an ensemble method designed to enhance predictive performance by combining multiple base models.

Formally, let $D = \{(x_i, y_i) \}_{i=1}^n$ represent the original dataset, where $x_i$ are the input features and $y_i$ are the target values.

First, $k$ base models $\{M_1, M_2, \ldots, M_k\}$ are trained on $D$, each learning to predict the target values $y_i$ from the input features $x_i$. Once trained, these base models generate predictions $\hat{y}_{ij} = M_j(x_i)$ for each instance $x_i$. These predictions are used to construct a new dataset $D_{meta} = \{(\hat{y}_{i1}, \hat{y}_{i2}, \ldots, \hat{y}_{ik}, y_i) \}_{i=1}^n$, where $\hat{y}_{ij}$ represents the prediction from the $j$-th base model for the $i$-th instance.

A meta-model $M_{meta}$ is then trained on $D_{meta}$. The inputs to $M_{meta}$ are the predictions from the base models $\{\hat{y}_{i1}, \hat{y}_{i2}, \ldots, \hat{y}_{ik}\}$, and the target values are $y_i$.

For a new instance $x_{new}$, the base models provide predictions $\{\hat{y}_{new,1}, \hat{y}_{new,2}, \ldots, \hat{y}_{new,k}\}$, which are then input to the meta-model $M_{meta}$ to produce the final prediction $\hat{y}_{new} = M_{meta}(\hat{y}_{new,1}, \hat{y}_{new,2}, \ldots, \hat{y}_{new,k})$.

This approach allows the meta-model to optimally combine the base models' predictions, thereby minimizing the generalization error and enhancing overall predictive performance.

Reply:
💀